packages/core-suggest/core-suggest.js

```diff
@@ -156,30 +157,30 @@ function onClick (self, event) {
 
 function onAjax (self) {
   if (!self.ajax) return
-  clearTimeout(onAjax.time) // Clear previous search
-  onAjax.ajax = onAjax.ajax || new window.XMLHttpRequest()
-  onAjax.ajax.abort() // Abort previous request
-  onAjax.time = setTimeout(onAjaxSend, AJAX_DEBOUNCE, self, onAjax.ajax) // Debounce
+  clearTimeout(self._xhrTime) // Clear previous search
+  self._xhr.abort() // Abort previous request
+  self._xhr.responseError = null
+  self._xhrTime = setTimeout(onAjaxSend, AJAX_DEBOUNCE, self) // Debounce
   return true
 }
 
-function onAjaxSend (self, ajax) {
+function onAjaxSend (self) {
   if (!self.input.value) return // Abort if input is empty
-  if (dispatchEvent(self, 'suggest.ajax.beforeSend', ajax)) {
-    ajax.onerror = () => dispatchEvent(self, 'suggest.ajax.error', ajax)
-    ajax.onload = () => {
-      if (ajax.status !== 200) return dispatchEvent(self, 'suggest.ajax.error', ajax)
+  if (dispatchEvent(self, 'suggest.ajax.beforeSend', self._xhr)) {
+    self._xhr.onerror = () => dispatchEvent(self, 'suggest.ajax.error', self._xhr)
```

Comment:
Should self._xhr.responseError be added here too?

Reply:
no, just in the bad json scenario!

Reply:
`xhr.responseError` is nice? :)

Reply:
i think `error` might be better since `xhr.onerror`s are not *response* errors..